libsolutil/JSON.cpp

```diff
@@ -104,46 +60,53 @@ void removeNullMembersHelper(Json::Value& _json)
 				removeNullMembersHelper(value);
 		}
 }
+#endif
 
 } // end anonymous namespace
 
-Json::Value removeNullMembers(Json::Value _json)
+Json removeNullMembers(Json _json)
 {
-	removeNullMembersHelper(_json);
+	// TODO: Support this.
+	// removeNullMembersHelper(_json);
 	return _json;
 }
 
-string jsonPrettyPrint(Json::Value const& _input)
+string jsonPrettyPrint(Json const& _input)
 {
 	return jsonPrint(_input, JsonFormat{ JsonFormat::Pretty });
 }
 
-string jsonCompactPrint(Json::Value const& _input)
+string jsonCompactPrint(Json const& _input)
 {
 	return jsonPrint(_input, JsonFormat{ JsonFormat::Compact });
 }
 
-string jsonPrint(Json::Value const& _input, JsonFormat const& _format)
+string jsonPrint(Json const& _input, JsonFormat const& _format)
 {
-	map<string, Json::Value> settings;
-	if (_format.format == JsonFormat::Pretty)
-	{
-		settings["indentation"] = string(_format.indent, ' ');
-		settings["enableYAMLCompatibility"] = true;
-	}
-	else
-		settings["indentation"] = "";
-	StreamWriterBuilder writerBuilder(settings);
-	string result = print(_input, writerBuilder);
-	if (_format.format == JsonFormat::Pretty)
-		boost::replace_all(result, " \n", "\n");
-	return result;
+	// NOTE: -1 here means no new lines (it is also the default setting)
+	return _input.dump(
+		/* indent */ (_format.format == JsonFormat::Pretty) ? static_cast<int>(_format.indent) : -1,
+		/* indent_char */ ' ',
+		/* ensure_ascii */ true
+	);
 }
 
-bool jsonParseStrict(string const& _input, Json::Value& _json, string* _errs /* = nullptr */)
+bool jsonParseStrict(string const& _input, Json& _json, string* _errs /* = nullptr */)
```

Comment:
Not necessarily in this PR but I think it would be better if this was returning either `optional<Json>` or maybe `variant<Json, string>`.